overlord/state/task.go

```diff
@@ -197,17 +197,21 @@ func (t *Task) Status() Status {
 }
 
 // SetStatus sets the task status, overriding the default behavior (see Status method).
-func (t *Task) SetStatus(new Status) {
+// Returns true if the Change status changed in response to this action.
+func (t *Task) SetStatus(new Status) bool {
 	t.state.writing()
 	old := t.status
 	t.status = new
 	if !old.Ready() && new.Ready() {
 		t.readyTime = timeNow()
 	}
 	chg := t.Change()
-	if chg != nil {
-		chg.taskStatusChanged(t, old, new)
+	if chg == nil {
+		return false
 	}
+	oldChangeStatus := chg.Status()
+	chg.taskStatusChanged(t, old, new)
+	return chg.Status() != oldChangeStatus
```

Comment:
now that I think more there are tasks that set themselves to Done,  it's not the case for the task we care about so I'm confused but this change probably is not useful enough and we can do something simpler. So we can leave SetStatus here alone

Reply:
OK. After some more debugging, I see that this does now work because `oldChangeStatus` was already set to `Done` (but I haven't yet understood why).